src/include/sof/ipc/common.h

```diff
@@ -90,6 +90,13 @@ struct ipc {
 #define ipc_get_drvdata(ipc) \
 	((ipc)->private)
 
+struct ipc_core_ctx {
+	int error;
```

Comment:
How in this design you are going to guarantee that ipc error code will not be overwritten prior to returning it in the reply message?

Reply:
@mwasko overwritten by whom? That variable is only set when processing IPC messages and a new IPC request cannot arrive before the previous one has been completed and replied to. This is still the case with IPC4, right? And asynchronous IPC messages, sent from the firmware to the host don't use that variable